vxdetector/VXdetector.py

```diff
@@ -4,246 +4,207 @@
+import Output_counter as Output_counter
+import files_manager as files_manager
+from interact_bowtie2 import mapbowtie2, buildbowtie2
+from interact_bedtools import overlap
+import tempfile
+from multiprocessing import Pool
+import biom
+import shutil
+import gzip
+import random
+
+def biom_to_fastq(biom_file, fastq_output):
+    '''Converts a BIOM file to FASTQ format by extracting sequences and ensuring they are compatible with FASTQ standards.'''
+    with open(fastq_output, "w") as f:
+        table = biom.load_table(biom_file)  # Load the BIOM table
+        sequences = table.ids(axis='observation')  # Extract sequences
+        for seq_id in sequences:
+            # Truncate the sequence ID if it’s too long for Bowtie2's requirements
+            truncated_id = seq_id[:50]  # Cut the sequence ID to 50 characters (adjust if needed)
+            # Check and sanitize the sequence to avoid unwanted characters
+            sanitized_seq = ''.join(filter(str.isalpha, seq_id))  # Ensure only letters are in the sequence
+            f.write(f"@{truncated_id}\n{sanitized_seq}\n+\n{'I' * len(sanitized_seq)}\n")  # Write in FASTQ format
+    print(f"BIOM file successfully converted to FASTQ format at: {fastq_output}")
+
+
+def sample_fastq(file_path, sample_size, sampled_indices=None):
+    '''Get random parts from the FASTQ file based on shared indices for paired-end reads.'''
+    sampled_reads = []  # List to hold sampled reads
+    open_func = gzip.open if file_path.endswith('.gz') else open
+
+    # Count total reads in the file by iterating line by line
+    with open_func(file_path, 'rt') as f:
+        total_reads = sum(1 for _ in f) // 4
+
+    # Adjust sample_size if it's greater than total_reads
+    if sample_size > total_reads:
+        sample_size = total_reads
+
+    if sampled_indices is None:
+        sampled_indices = sorted(random.sample(range(total_reads), sample_size))
+
+    with open_func(file_path, 'rt') as f:
+        read_idx = 0
+        read_buffer = []
+        for line in f:
+            read_buffer.append(line)
+            if len(read_buffer) == 4:  # Once we have a full read
+                if read_idx in sampled_indices:
+                    sampled_reads.extend(read_buffer)  # Add the read if it matches the sampled index
+                read_buffer = []  # Clear the buffer for the next read
+                read_idx += 1  # Move to the next read
+
+                # Stop if we've collected enough reads
+                if len(sampled_reads) >= sample_size * 4:
+                    break
+
+    return sampled_reads, sampled_indices
+
+
+def save_sampled_fastq(sampled_reads, output_path):
+    '''Save sampled reads to a temporary FASTQ file.'''
+    open_func = gzip.open if output_path.endswith('.gz') else open  # Use gzip if output is .gz
+
+    with open_func(output_path, 'wt') as f:  # Write in text mode ('wt')
+        f.writelines(sampled_reads)
 
 
 def do_statistic(result):
```

Comment:
it's hard for me to see what these changes are good for? Is this "just" refactoring the code OR also changing the functionality. You might want to make sure the github actions pass as they should include some unit tests